Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -408,30 +408,69 @@ public final class ManifestLoader: ManifestLoaderProtocol {
     private func validateBinaryTargets(_ manifest: Manifest, observabilityScope: ObservabilityScope) throws {
         // Check that binary targets point to the right file type.
         for target in manifest.targets where target.type == .binary {
-            guard let location = URL(string: target.url ?? target.path ?? "") else {
+            if target.isLocal {
+                guard let path = target.path else {
+                    observabilityScope.emit(.invalidBinaryLocation(targetName: target.name))
+                    continue
+                }
+
+                guard let path = path.spm_chuzzle(), !path.isEmpty else {
+                    observabilityScope.emit(.invalidLocalBinaryPath(path: path, targetName: target.name))
+                    continue
+                }
+
+                guard let relativePath = try? RelativePath(validating: path) else {
```

Comment:
Should we make this a do-catch and include the underlying error?

Reply:
this is intentional - the underlying error in this case would be a generic "the path <path> is an invalid relative path" which we fine-tune to invalidLocalBinaryPath which explains that this must be relative to the project root

Reply:
Right, the question is what happens if the underlying error ever changes and includes useful info. I guess that's not super likely in this particular case.

Reply:
right, that was my thinking here as it basically validates that the value is valid relative path